src/topp/GNPSExport.cpp

```diff
@@ -0,0 +1,351 @@
+
+    registerInputFileList_("in_mzml", "<files>", ListUtils::create<String>(""), "original mzml files containing ms/ms spectrum information");
+    setValidFormats_("in_mzml", ListUtils::create<String>("mzML"));
+
+    registerOutputFile_("out", "<file>", "", "Output MGF file");
+    setValidFormats_("out", ListUtils::create<String>("mgf"));
+
+    registerStringOption_("output_type", "<choice>", "full_spectra", "specificity of mgf output information", false);
+    setValidStrings_("output_type", ListUtils::create<String>("full_spectra,merged_spectra"));
+
+    registerDoubleOption_("precursor_mz_tolerance", "<num>", DEF_PRECURSOR_MZ_TOLERANCE, "Tolerance mz window for precursor selection", false);
+    registerDoubleOption_("precursor_rt_tolerance", "<num>", DEF_PRECURSOR_RT_TOLERANCE, "Tolerance rt window for precursor selection", false);
+
+    registerTOPPSubsection_("merged_spectra", "Options for exporting mgf file with merged spectra per feature");
+    registerDoubleOption_("merged_spectra:cos_similarity", "<num>", DEF_COSINE_SIMILARITY, "Cosine similarity threshold for merged_spectra output", false);
+  }
+
+  // the main function is called after all parameters are read
+  ExitCodes main_(int, const char **)
+  {
+    ProgressLogger progress_logger;
+    progress_logger.setLogType(log_type_);
+
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+    String consensus_file_path(getStringOption_("in_cm"));
+    StringList mzml_file_paths(getStringList_("in_mzml"));
+
+    String out(getStringOption_("out"));
+
+    String output_type(getStringOption_("output_type"));
+
+    double prec_mz_tol(getDoubleOption_("precursor_mz_tolerance"));
+    double prec_rt_tol(getDoubleOption_("precursor_rt_tolerance"));
+
+    double cos_sim(getDoubleOption_("merged_spectra:cos_similarity"));
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+    // ConsensusMap
+    ConsensusXMLFile consensus_file;
+    consensus_file.setLogType(log_type_);
+    ConsensusMap consensus_map;
+    consensus_file.load(consensus_file_path, consensus_map);
+
+    // MSExperiment
+    vector<MSExperiment> ms_maps;
+    for(auto mzml_file_path : mzml_file_paths) {
+      MzMLFile mzml_file;
+      MSExperiment map;
+      mzml_file.setLogType(log_type_);
+      mzml_file.load(mzml_file_path, map);
+
+      ms_maps.push_back(map);
+    }
+
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+    progress_logger.startProgress(0, consensus_map.size(), "parsing features and ms2 identifications...");
+    std::stringstream output_stream;
+    Size feature_count = 1;
+    for(Size i = 0; i != consensus_map.size(); ++i) {
```

Comment:
can you change the style to
```
for (...)
{
}
```

and follow our https://github.com/OpenMS/OpenMS/wiki/Coding-conventions